### PR TITLE
feat: Add `addAll`, `updateAll` and `removeAll` methods

### PR DESCRIPTION
**Pull request by wolfenrain**

### CODE REVIEW

File: packages/deposit/lib/src/adapters/memory_deposit_adapter.dart [hunk 1]
```diff
@@ -7,26 +7,44 @@ class MemoryDepositAdapter extends DepositAdapter<int> {
...
     _ref(table).add(result);
     return result;
   }
 
+  @override
+  Future<List<Map<String, dynamic>>> addAll(
+    String table,
+    String primaryColumn,
+    List<Map<String, dynamic>> data,
+  ) async {
```
Comment (erickzanardo) [inline packages/deposit/lib/src/adapters/memory_deposit_adapter.dart]:
```suggestion
  ) {
```

File: packages/deposit/lib/src/adapters/memory_deposit_adapter.dart [hunk 2]
```diff
@@ -83,7 +101,18 @@ class MemoryDepositAdapter extends DepositAdapter<int> {
...
+      (d) => d[primaryColumn] == data[primaryColumn],
+    );
+  }
+
+  @override
+  Future<void> removeAll(
+    String table,
+    String primaryColumn,
+    List<Map<String, dynamic>> data,
+  ) async {
```
Comment (erickzanardo) [inline packages/deposit/lib/src/adapters/memory_deposit_adapter.dart:114]:
```suggestion
  ) {
```

File: packages/deposit/lib/src/adapters/memory_deposit_adapter.dart [hunk 3]
```diff
@@ -92,6 +121,20 @@ class MemoryDepositAdapter extends DepositAdapter<int> {
...
+    );
+    return result..addAll(data);
+  }
+
+  @override
+  Future<List<Map<String, dynamic>>> updateAll(
+    String table,
+    String primaryColumn,
+    List<Map<String, dynamic>> data,
+  ) async {
```
Comment (erickzanardo) [inline packages/deposit/lib/src/adapters/memory_deposit_adapter.dart]:
```suggestion
  ) {
```

File: packages/deposit_firestore/lib/src/firestore_deposit_adapter.dart [hunk 4]
```diff
@@ -37,6 +37,15 @@ class FirestoreDepositAdapter extends DepositAdapter<String> {
     }
   }
 
+  @override
+  Future<List<Map<String, dynamic>>> addAll(
+    String table,
+    String primaryColumn,
+    List<Map<String, dynamic>> data,
+  ) async {
```
Comment (erickzanardo) [inline packages/deposit_firestore/lib/src/firestore_deposit_adapter.dart]:
```suggestion
  ) {
```

File: packages/deposit/test/memory_deposit_adapter_test.dart [hunk 5]
```diff
@@ -21,10 +21,30 @@ void main() {
...
-        expect(result?.first['brand'], equals('VW'));
-        expect(result?.first['model'], equals('Nivus'));
+        expect(result!.length, equals(1));
+        expect(result.first['brand'], equals('VW'));
+        expect(result.first['model'], equals('Nivus'));
+      });
+    });
+
+    group('.addAll()', () {
+      test('add multiple items', () async {
```
Comment (erickzanardo) [inline packages/deposit/test/memory_deposit_adapter_test.dart]:
```suggestion
      test('adds multiple items', () async {
```

File: packages/deposit/test/memory_deposit_adapter_test.dart [hunk 6]
```diff
@@ -12,7 +12,7 @@ void main() {
     });
 
     group('.add()', () {
-      test('should add a single item', () async {
+      test('add a single item', () async {
```
Comment (erickzanardo) [inline packages/deposit/test/memory_deposit_adapter_test.dart]:
```suggestion
      test('adds a single item', () async {
```

File: packages/deposit_firestore/lib/src/firestore_deposit_adapter.dart [hunk 7]
```diff
@@ -95,6 +104,15 @@ class FirestoreDepositAdapter extends DepositAdapter<String> {
     await doc?.delete();
   }
 
+  @override
+  Future<void> removeAll(
+    String table,
+    String primaryColumn,
+    List<Map<String, dynamic>> data,
+  ) async {
```
Comment (erickzanardo) [inline packages/deposit_firestore/lib/src/firestore_deposit_adapter.dart:112]:
```suggestion
  ) {
```

File: packages/deposit_firestore/lib/src/firestore_deposit_adapter.dart [hunk 8]
```diff
@@ -108,10 +126,19 @@ class FirestoreDepositAdapter extends DepositAdapter<String> {
     return data;
   }
 
+  @override
+  Future<List<Map<String, dynamic>>> updateAll(
+    String table,
+    String primaryColumn,
+    List<Map<String, dynamic>> data,
+  ) async {
```
Comment (erickzanardo) [inline packages/deposit_firestore/lib/src/firestore_deposit_adapter.dart]:
```suggestion
  ) {
```

File: packages/deposit_firestore/test/firestore_deposit_test.dart [hunk 9]
```diff
@@ -167,5 +213,41 @@ void main() {
         expect(docs.length, equals(1));
       });
     });
+
+    group('.updateAll()', () {
+      test('update multiple items', () async {
```
Comment (erickzanardo) [inline packages/deposit_firestore/test/firestore_deposit_test.dart]:
```suggestion
      test('updates multiple items', () async {
```

File: packages/deposit_supabase/lib/src/supabase_deposit_adapter.dart [hunk 10]
```diff
@@ -90,6 +99,15 @@ class SupabaseDepositAdapter extends DepositAdapter<int> {
     }
   }
 
+  @override
+  Future<void> removeAll(
+    String table,
+    String primaryColumn,
+    List<Map<String, dynamic>> data,
+  ) async {
```
Comment (erickzanardo) [inline packages/deposit_supabase/lib/src/supabase_deposit_adapter.dart:107]:
```suggestion
  ) {
```

File: packages/deposit_supabase/lib/src/supabase_deposit_adapter.dart [hunk 11]
```diff
@@ -105,4 +123,13 @@ class SupabaseDepositAdapter extends DepositAdapter<int> {
     }
     return (response.data as List<dynamic>).cast<Map<String, dynamic>>().first;
   }
+
+  @override
+  Future<List<Map<String, dynamic>>> updateAll(
+    String table,
+    String primaryColumn,
+    List<Map<String, dynamic>> data,
+  ) async {
```
Comment (erickzanardo) [inline packages/deposit_supabase/lib/src/supabase_deposit_adapter.dart]:
```suggestion
  ) {
```

File: packages/deposit_supabase/test/supabase_deposit_adapter_test.dart [hunk 12]
```diff
@@ -51,6 +51,28 @@ void main() {
       });
     });
 
+    group('.addAll()', () {
+      test('add multiple items', () async {
```
Comment (erickzanardo) [inline packages/deposit_supabase/test/supabase_deposit_adapter_test.dart]:
```suggestion
      test('adds multiple items', () async {
```

File: packages/deposit_supabase/test/supabase_deposit_adapter_test.dart [hunk 13]
```diff
@@ -51,6 +51,28 @@ void main() {
...
+          'cars',
+          'id',
+          [
+            <String, dynamic>{'brand': 'VW', 'model': 'Nivus'},
+            <String, dynamic>{'brand': 'VW', 'model': 'Virtus'},
+          ],
+        );
+
+        final result = await client.from('cars').select().execute();
+        final data = (result.data as List).cast<Map>();
```
Comment (erickzanardo) [inline packages/deposit_supabase/test/supabase_deposit_adapter_test.dart:66]: This line looks weird, you first cast to `List` then to a `Map`?  Am I am missing something hahaha?

File: packages/deposit_supabase/test/supabase_deposit_adapter_test.dart [hunk 14]
```diff
@@ -207,5 +248,36 @@ void main() {
         expect(result.first['id'], equals(data['id']));
       });
     });
+
+    group('.updateAll()', () {
+      test('update multiple items', () async {
```
Comment (erickzanardo) [inline packages/deposit_supabase/test/supabase_deposit_adapter_test.dart]:
```suggestion
      test('updates multiple items', () async {
```

**Review by erickzanardo — APPROVED** [review]
One doubt and a bunch of nits, but LGTM already.

File: packages/deposit/lib/src/adapters/memory_deposit_adapter.dart (hunk 2 above)
Comment (wolfenrain) [reply packages/deposit/lib/src/adapters/memory_deposit_adapter.dart:114]: This one has to stay `async` cause the `Future.wait` is not returned, but you do want to wait until it is finished

File: packages/deposit_firestore/lib/src/firestore_deposit_adapter.dart (hunk 7 above)
Comment (wolfenrain) [reply packages/deposit_firestore/lib/src/firestore_deposit_adapter.dart:112]: This one has to stay async cause the Future.wait is not returned, but you do want to wait until it is finished

File: packages/deposit_supabase/lib/src/supabase_deposit_adapter.dart (hunk 10 above)
Comment (wolfenrain) [reply packages/deposit_supabase/lib/src/supabase_deposit_adapter.dart:107]: This one has to stay async cause the Future.wait is not returned, but you do want to wait until it is finished

File: packages/deposit_supabase/test/supabase_deposit_adapter_test.dart (hunk 13 above)
Comment (wolfenrain) [reply packages/deposit_supabase/test/supabase_deposit_adapter_test.dart:66]: It is a convoluted way to cast a `List<Map>`. If I did it as `result.data as List<Map>` it complains that a `List<dynamic>` cant be cast too `List<Map>`.

**Pull request merged**